### PR TITLE
feat: add campaign monitor usage report

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds usage report for Campaign Monitor

### How to test the changes in this Pull Request:

1. Make sure you have the Campaign Monitor credentials on Newsletters setting to an account that has some activity
2. `wp shell`
3.`$x = Newspack_Newsletters_Campaign_Monitor::instance()->get_usage_report();`
4. `$x->to_array();`

Make sure you get an output similar to:

```
=> array(8) {
  ["date"]=>
  string(10) "2023-11-14"
  ["emails_sent"]=>
  int(0)
  ["opens"]=>
  int(0)
  ["clicks"]=>
  int(0)
  ["unsubscribes"]=>
  int(0)
  ["subscribes"]=>
  int(2)
  ["total_contacts"]=>
  int(6)
  ["growth_rate"]=>
  float(0.3333333333333333)
}
```

Campaigns data are calculated with the difference between what happened between now and the last time the check was performed (this report is supposed to be fetched once a day)

If you have some Campaigns with activity, the first time you run the script emails_sent, clicks and open will all come with `0`.

If you want to test how the script would behave the next day, edit the values on the `'np_newsletters_campaign_data_' . $campaign_id` option. Set it to some values lower than what you have now and the report should output the difference.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+		if ( is_wp_error( $campaigns ) ) {
+			return $campaigns;
+		}
+		return new Newspack_Newsletters_Service_Provider_Usage_Report( array_merge( $subs, $campaigns ) );
+	}
+
+	/**
+	 * Get subscribers and unsubscribers for the last $days days.
+	 *
+	 * @return array|WP_Error Array of subscribers and unsubscribers for the last $days days, or WP_Error on failure.
```
Comment (adekbadek) [inline includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php:84]: Function comment mentions `$days`, but it's not in the function body.

File: includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+		$campaign_data = [
+			'opens'  => $summary->UniqueOpened,
+			'clicks' => $summary->Clicks,
+		];
+
+		$campaign_last_data = get_option( $campaign_last_data_option_name );
+		update_option( $campaign_last_data_option_name, $campaign_data );
+
+		if ( $on_target_day ) {
+			// This campaign was just sent. All the data refers from yesterday.
```
Comment (adekbadek) [inline includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php:235]:
```suggestion
			// This campaign was just sent. All the data is from yesterday.
```

**Review by adekbadek — APPROVED** [review]
Left non-blocking comments.

File: includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php (hunk 1 above)
Comment (leogermani) [reply includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php:84]: thanks

Comment (leogermani) [reply includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php:84]: wow and there's a trailing comma in the arguments list: `get_subscribers_for_list( $list_id, ) {`

fixing it now

**Pull request merged**